packages/block-library/src/archives/edit.js

```diff
@@ -4,24 +4,51 @@
 import {
 	ToggleControl,
 	SelectControl,
-	Disabled,
+	Spinner,
 	__experimentalToolsPanel as ToolsPanel,
 	__experimentalToolsPanelItem as ToolsPanelItem,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { InspectorControls, useBlockProps } from '@wordpress/block-editor';
-import ServerSideRender from '@wordpress/server-side-render';
+import { useServerSideRender } from '@wordpress/server-side-render';
+import { useDisabled } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
 import { useToolsPanelDropdownMenuProps } from '../utils/hooks';
+import HtmlRenderer from '../utils/html-renderer';
 
-export default function ArchivesEdit( { attributes, setAttributes } ) {
+export default function ArchivesEdit( { attributes, setAttributes, name } ) {
 	const { showLabel, showPostCounts, displayAsDropdown, type } = attributes;
 
 	const dropdownMenuProps = useToolsPanelDropdownMenuProps();
 
+	const { content, status, error } = useServerSideRender( {
+		attributes,
+		skipBlockSupportAttributes: true,
+		block: name,
+	} );
+
+	const disabledRef = useDisabled();
+	const blockProps = useBlockProps( { ref: disabledRef } );
```

Comment:
In addition to the `wp-block-archives` class, this block will have one of the following classes added:

However, the `wrapperProps` prop will overwrite the class name generated by server-side rendering.

Therefore, we need to explicitly add this class:

```javascript
const blockProps = useBlockProps( {
	ref: disabledRef,
	className: clsx( 'wp-block-archives', {
		'wp-block-archives-dropdown': displayAsDropdown,
		'wp-block-archives-list': ! displayAsDropdown,
	} ),
} );
```

| Before | After |
|--------|--------|
| <img width="586" height="309" alt="Image" src="https://github.com/user-attachments/assets/003a7ab3-8475-4640-8247-ddc23afc9e9c" /> | <img width="573" height="308" alt="Image" src="https://github.com/user-attachments/assets/df1ed733-c65d-4201-8fd5-af77253b2cd0" /> |